functions/Set-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,302 @@
+Changes the name of the step in "Job1" with the name "Step1" to "Step2"
+
+.EXAMPLE   
+Set-DbaAgentJobStep -SqlInstance 'sql' -JobName 'Job1' -StepName 'Step1' -Database 'msdb'
+Changes the database of the step in "Job1" with the name "Step1" to 'msdb'
+
+.EXAMPLE   
+Set-DbaAgentJobStep -SqlInstance "sql1", "sql2", "sql3" -JobName 'Job1' -StepName 'Step1' -DatabaseName 'msdb'
+Changes the database of the step in "Job1" with the name "Step1" to 'msdb' for multiple servers
+
+.EXAMPLE   
+"sql1", "sql2", "sql3" | Set-DbaAgentJobStep -JobName 'Job1' -StepName 'Step1' -DatabaseName 'msdb'
+Changes the database of the step in "Job1" with the name "Step1" to 'msdb' for multiple servers using pipe line
+
+#>   
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = "Low")]
+
+    param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer")]
+        [object[]]$SqlInstance,
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$JobName,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$StepName,
+        [Parameter(Mandatory = $false)]
+        [string]$NewName,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('ActiveScripting', 'AnalysisCommand', 'AnalysisQuery', 'CmdExec', 'Distribution', 'LogReader', 'Merge', 'PowerShell', 'QueueReader', 'Snapshot', 'Ssis', 'TransactSql')]
+        [string]$Subsystem,
+        [Parameter(Mandatory = $false)]
+        [string]$Command,
+        [Parameter(Mandatory = $false)]
+        [int]$CmdExecSuccessCode,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('QuitWithSuccess', 'QuitWithFailure', 'GoToNextStep', 'GoToStep')]
+        [string]$OnSuccessAction,
+        [Parameter(Mandatory = $false)]
+        [int]$OnSuccessStepId,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('QuitWithSuccess', 'QuitWithFailure', 'GoToNextStep', 'GoToStep')]
+        [string]$OnFailAction,
+        [Parameter(Mandatory = $false)]
+        [int]$OnFailStepId,
+        [Parameter(Mandatory = $false)]
+        [string]$DatabaseName,
+        [Parameter(Mandatory = $false)]
+        [string]$DatabaseUserName,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryAttempts,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryInterval,
+        [Parameter(Mandatory = $false)]
+        [string]$OutputFileName,
+        [Parameter(Mandatory = $false)]
+        [string]$ProxyName,
+        [switch]$Silent
+    )
+
+    begin{
+        # Check the parameter on success step id
+        if (($OnSuccessAction -ne 'GoToStep') -and ($OnSuccessStepId -ge 1)) {
+            Stop-Function -Message "Parameter OnSuccessStepId can only be used with OnSuccessAction 'GoToStep'."  -Target $instance 
```

Comment:
Not processing any item yet, so `-Target $Instance` is not appropriate yet (and the variable $instance doesn't exist yet).